### ISSUE TITLE
Navigation Menu with Auto-Start-Level

### ISSUE THREAD

**Issue by Hendriks** [contao/core#8228]
What about an extra option for the frontend navigation module to set an **auto-start-level**?

I think a frontend nav module which shows **only child pages of the active parent** is something quite oftenly required, at least for websites with larger / deeper page structures... Shouldn't be too hard to implement this as a **core feature**; I think (but not know ;) Don't you think?

Greets, Hendriks

**Comment by asaage** [comment]
+1 for such a feature  - [like that](https://community.contao.org/de/showthread.php?47199-Navigation-mit-Seiten-von-gleicher-Ebene&p=390988&viewfull=1#post390988)

**Comment by Toflar** [comment]
Do I understand this correctly that it would behave just like if you would check the checkbox `Set a reference page` and then instead of having to choose one, it would simply always take the current page as `Reference page`?

**Comment by Hendriks** [comment]
@Toflar: Yes, this sounds like the kind of pretty simple solution I was imagining... This would completely do the job, I guess... So it could be just another checkbox underneath "set reference page" called "reference to current"...

**Comment by Toflar** [comment]
That's very easy, I'll prepare a Pull Request for Contao 4.2.

**Comment by Hendriks** [comment]
Didn't we have this issue (request) beforehand? And you say it's sooo easy... :) Why not implement it for 3.5 then as well? Shouldn't break existing installations, a checkbox which will be "off" by defaut...

**Comment by Toflar** [comment]
Why do we have to answer this question over and over again? Contao 3.5 is feature complete. Bugfix releases never contain any features and **nothing** will make it into 3.5 anymore except for bugfixes.

**Comment by Hendriks** [comment]
Oh... I wasn't aware / thinking of the current state of 3.5... Sorry...

**Comment by fritzmg** [comment]
On that not, it would be nice if Contao also supports the start- and stop-level when a reference page is set in its default navigation module in some future version.

**Comment by Toflar** [comment]
Actually, it does support the stop-level, not the start level though. Is that intended or is that a bug @leofeyer?

**Comment by leofeyer** [comment]
There is no start level when there is a reference page, because the reference page is the start level.

**Comment by Toflar** [comment]
Which makes no sense :smile: Why should I not start on level 2 based on the reference page? :smile: So to be BC, we would need yet another checkbox :smile:

**Comment by leofeyer** [comment]
And probably more smileys :smile:

**Comment by Toflar** [comment]
Yes :+1: Enjoy weekend, guys!

**Comment by Toflar** [comment]
So I had a look at this and I have to admit, after reading the code I cannot see the initial problem here anymore. I do see the need for a checkbox "Consider start level" when you set a reference page, because that's currently not possible.
However, you ask for a

> nav module which shows only child pages of the active parent

This is the default behaviour and it has been ever since? If you do not set a `reference page` it will always take the current page as reference? Am I too stupid to understand now? Can anybody help me out?

**Comment by fritzmg** [comment]
> This is the default behaviour and it has been ever since? If you do not set a reference page it will always take the current page as reference? Am I too stupid to understand now? Can anybody help me out?

No, by default it shows the whole tree.

When you do not set a reference page or a start level, the following happens in ModuleNavigation:
```php
$trail = $objPage->trail;
$level = ($this->levelOffset > 0) ? $this->levelOffset : 0;
```
```php
$this->Template->items = $this->renderNavigation($trail[$level], 1, $host, $lang);
```
So `$trail` will be an array containing the parent page IDs of the current page (`$objPage`). When you do not set a start level, `$level` will be `0`. The function `Module::renderNavigation` gets `$trail[$level]` as the first parameter, which is the parameter `$pid`, i.e. the page ID of the start page, i.e. reference page. Since `$level` is `0`, `$trail[$level]` will return the page ID of the root page.
by default for the

**Comment by Toflar** [comment]
I can read the code myself, I needed clarification of the goal :) But I found what I did not read: The **parent** of the current page! I read something like "use the current page as reference" which **is** the default behaviour. But actually when I read this now, it's like support entering negative values in `start level`, isn't it? So

> I think a frontend nav module which shows only child pages of the active parent

actually translates to a `start level` of `-1` (parent level), right?
So given a tree like this:

* News
    * 2016
        * **Test**
    * 2015
    * 2014

Given I am on the page `Test` I could get the following result by setting `start level` to `-1` and `stop level` to `-1`:

* 2016
* 2015
* 2016

And I could get the following result by setting `start level` to `-1` and `stop level` to `0`:

* 2016
    * **Test**
* 2015
* 2016

@Hendriks is that correct?

**Comment by fritzmg** [comment]
No, I think the desired behavior is to have a Navigation Module, that always sets the reference page to the current page, no matter how deep that page is in the page tree.

If you do it with start levels again, then the modul is again fixed to a specific level in the page tree (relative to the current page).

**Comment by frontendschlampe** [comment]
This feature is implemented in our navigation https://github.com/backbone87/contao-navigation.git - for Contao 3.5.x, too!

**Comment by Toflar** [comment]
Then I don't get this issue, I'm sorry. Please feel free to work on it and submit a PR, if you get it, that's probably easier :-)

**Comment by fritzmg** [comment]
Just look at the code [here](https://community.contao.org/de/showthread.php?47199-Navigation-mit-Seiten-von-gleicher-Ebene&p=390988&viewfull=1#post390988) ;). 
All you need is
```php
global $objPage;
$objModuleModel->rootPage = $objPage->pid; 
```
Then the navigation module will always show only the children of the current page.

**Comment by asaage** [comment]
This will make the currentPage the rootPage of the nav right?
This will still not allow the parentPage to be the rootPage of the nav as the startlevel-number is relative to the Website-Root and negative numbers are not supported right?

**Comment by fritzmg** [comment]
> This will make the currentPage the rootPage of the nav right?
> This will still not allow the parentPage to be the rootPage of the nav as the startlevel-number is relative to the Website-Root and negative numbers are not supported right?

Ah, you are right, i did misunderstand @Hendriks original issue a bit. I thought he wanted `$objPage->id` as the root, but he wants actually the parent, i.e. `$objPage->pid` (as in my original code). Sorry for the confusion.

**Comment by asaage** [comment]
another point is:
if you set a stopLevel the navigation will disappear if you are deep enough in the tree
